src/uni_ai_chatbot/main.py

```diff
@@ -1,40 +1,48 @@
 import os
 from pathlib import Path
-
 from langchain.chains import RetrievalQA
 from langchain_community.document_loaders import TextLoader
 from langchain_community.vectorstores import FAISS
-from langchain_ollama import OllamaEmbeddings
-from langchain_ollama import OllamaLLM
 from langchain_text_splitters import CharacterTextSplitter
+from langchain_mistralai import MistralAIEmbeddings
+from langchain_mistralai import ChatMistralAI
+
 
 from uni_ai_chatbot.resources import get_resource
 
-# Get Ollama host from environment variable or use default
-ollama_host = os.environ.get("OLLAMA_HOST", "http://localhost:11434")
+# Load API key from env (make sure it's set)
+mistral_api_key = os.environ.get("MISTRAL_API_KEY")
+if not mistral_api_key:
```

Comment:
This can be nicely encapsulated using `pydantic-settings` 

```python
from pydantic_settings import BaseSettings
class MyChainSettings(BaseSettings):
    MISTRAL_API_KEY: str  # will look for MISTRAL_API_KEY env var on instantiating MyChainSettings and will fail with explaining error if it's missing 
    HF_TOKEN: str | None = None  # optional, as has default value

```